### PR TITLE
Upgrade Sulu 2.0

### PR DESCRIPTION
**Pull request by markusmoosbrugger**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | 
| Deprecations? | no
| Fixed tickets | part of #190
| Related issues/PRs | #issuenum
| License | MIT

#### What's in this PR?

- Implement new list of forms
- Implement form for form details (form fields are static at the moment)
- Export single_form_selection content type

#### Why?

Upgrade to work with Sulu 2.0

#### BC Breaks/Deprecations

The content type **single_form_selection** has to be used in a different way than the content type **form_select** previously. The param name "type" has changed to "resourceKey" because "type" is overwritten internally to specify the type of the content type (single_select in this case) itself.

Example Usage:
```xml
<property name="form" type="single_form_selection">
     <meta>
         <title lang="de">Formular</title>
         <title lang="en">Form</title>
     </meta>
     <params>
         <param name="resourceKey" value="page"/>
      </params>
</property>
```

#### To Do

- [ ] Update CHANGELOG.md
- [ ] Create documentation

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -3,16 +3,16 @@
     "description": "Bundle for creating forms in Sulu.",
     "type": "sulu-bundle",
     "require": {
-        "symfony/symfony": "^2.8.7 || ^3.0",
-        "symfony/swiftmailer-bundle": "^2.3 || ^3.0",
+        "symfony/symfony": "^2.8.7 || ^3.0 || ^4.0",
```
Comment (alexander-schranz) [inline composer.json]: we should not longer require  the whole `symfony/symfony` framework here we should replace this with:

```
        "symfony/config": "^3.4 || ^4.0",
        "symfony/dependency-injection": "^3.4 || ^4.0",
        "symfony/event-dispatcher": "^3.4 || ^4.0",
        "symfony/form": "^3.4 || ^4.0",
        "symfony/validator": "^3.4 || ^4.0",
```

File: composer.json [hunk 2]
```diff
@@ -3,16 +3,16 @@
...
+        "symfony/swiftmailer-bundle": "^2.3 || ^3.0 || ^4.0",
```
Comment (alexander-schranz) [inline composer.json]: it seems the swiftmailer is versionated not the same way as symfony so we need remove ^4.0

```
        "symfony/swiftmailer-bundle": "^2.3 || ^3.0",
```

File: Admin/FormAdmin.php [hunk 3]
```diff
@@ -75,9 +156,15 @@ public function getSecurityContexts()
         return [
             'Sulu' => [
                 'Form' => [
-                    'sulu.form.forms',
+                    'sulu.form.forms' => [
```
Comment (alexander-schranz) [inline Admin/FormAdmin.php]: /cc @chirimoya this seems to be a BC Break that it need that permissions need to be defined

File: Resources/config/form-fields/header.xml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        </meta>
+        <params>
+            <param name="label">
+                <meta>
+                    <title>sulu_form.width</title>
+                </meta>
+            </param>
+            <param name="default_value" value="0"/>
+            <param name="values" type="collection">
+                <param name="0">
```
Comment (kleinkoerkamp) [inline Resources/config/form-fields/header.xml]: Name should be 'full' instead of '0' so that's compatible with Form/Type/DynamicFormType.php

Comment (markusmoosbrugger) [reply Resources/config/form-fields/header.xml]: Fixed. Thanks for pointing this out.

File: Dynamic/Types/RecaptchaType.php [hunk 5]
```diff
@@ -30,7 +30,7 @@ public function getConfiguration()
     {
         return new FormFieldTypeConfiguration(
             'sulu_form.type.recaptcha',
-            'SuluFormBundle:forms:fields/types/recaptcha.html.twig',
+            'SuluFormBundle/Resources/config/form-fields/default_field.xm',
```
Comment (alexander-schranz) [inline Dynamic/Types/RecaptchaType.php]: .xml

File: Dynamic/Types/MailchimpType.php [hunk 6]
```diff
@@ -42,7 +42,7 @@ public function getConfiguration()
     {
         return new FormFieldTypeConfiguration(
             'sulu_form.type.mailchimp',
-            'SuluFormBundle:forms:fields/types/mailchimp.html.twig',
+            'SuluFormBundle/Resources/config/form-fields/field_mailchimp.xm',
```
Comment (alexander-schranz) [inline Dynamic/Types/MailchimpType.php]: .xml

File: Resources/config/services.xml [hunk 7]
```diff
@@ -66,21 +58,21 @@
...
-                 class="Sulu\Bundle\FormBundle\Content\Types\FormSelect">
+        <service id="sulu_form.content_type.single_form_selection"
+                 class="Sulu\Bundle\FormBundle\Content\Types\SingleFormSelection">
             <argument>SuluFormBundle:Template:content-types/form_select_simple.html.twig</argument>
             <argument type="service" id="sulu_form.repository.form" />
             <argument type="service" id="sulu_form.builder" />
             <argument type="service" id="sulu_form.reference_store.form"/>
 
-            <tag name="sulu.content.type" alias="form_select"/>
+            <tag name="sulu.content.type" alias="single_form_selection"/>
```
Comment (alexander-schranz) [inline Resources/config/services.xml]: Add the following to the UPGRADE.md:

~~~~
# dev-develop

The content type for form selection has been changed from `form_select` to `single_form_selection` also the param `type` has changed to `resourceKey`:

**Before**

```xml
<property name="form" type="form_select">
    <params>
        <param name="type" value="page" />
    </params>
</property>
```

**Before**

```xml
<property name="form" type="single_form_selection">
    <params>
        <param name="resourceKey" value="page" />
    </params>
</property>
```
~~~~

**Comment by alexander-schranz** [comment]
We will merge this state into develop branch to keep the PR small as possible.

**Pull request merged**